src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -385,6 +422,33 @@ def _maybe_adjust(self):
                     'var': 'pg_num',
                     'val': str(p['pg_num_final'])
                 })
+                
+                # Create new event for each pool
+                # and update existing events
+                # Call Progress Module to create progress event
+                if pool_id not in self._event:
+                    osdmap = self.get_osdmap()
+                    pools = osdmap.get_pools()
+                    pool_data = pools[int(pool_id)]
+                    pg_num = pool_data['pg_num']
+                    pg_num_target = pool_data['pg_num_target']
+                    ev_id = str(uuid.uuid4())
+                    if pg_num < pg_num_target:
+                        pg_adj_obj = PgAdjustmentProgress(pg_num, pg_num_target, ev_id, 'increasing')
```

Comment:
now only pg_adj_obj construction is different between these branches - the other two calls can be after the if/else